engine/class_modules/sc_death_knight.cpp

```diff
@@ -3004,6 +3021,41 @@ struct death_knight_melee_attack_t : public death_knight_action_t<melee_attack_t
 
   void trigger_icecap( const action_state_t* state ) const;
   void trigger_razorice( const action_state_t* state ) const;
+
+  double cost() const override
+  {
+    double c = death_knight_action_t::cost();
+    if ( this -> current_resource() == RESOURCE_RUNE )
+    {
+      // If we spend a rune, we have a chance to spread the dot
+      if ( this->triggers_shackle_the_unworthy && as<int>(c) > 0 && p() -> covenant.shackle_the_unworthy -> ok() && p() -> cooldown.shackle_the_unworthy_icd -> is_ready() )
```

Comment:
I think execute() is a better place for this. Place it after the action_t::execute call, and you can access last_resource_cost to get the last cost used.

Reply:
Serge is right, cost() is definitely not its place.

I have a few things to add:
If it's the same code for death_knight_melee_attack_t and death_knight_spell_t, you could just override death_knight_action_t::execute(). You could also add it inside death_knight_t::resource_loss() as it receives the action triggering it (although you will have to type it to death_knight_action_t to get the shackles boolean).

Does it also proc on rune spenders when their cost is free (Howling Blast with Rime, DRW abilities?).

Can the dot trigger itself on a target that was already affected before (ie. can you bounce the dot back between two targets indefinitely?)

Reply:
@Melekus Yes it can proc on reduce cost/free rune spenders. It also spreads at its current remaining time (mostly, there is a few ms variance in spread duration...), never spreading to a target its already active on, so its impossible to maintain it indefinitely.

Reply:
Tested with howling blast w/ rime and it definately does spread.  DRW I couldn't tell.  It attributes the buff refreshing by me and comes after my attacks, but before drw's clone attack, so it looks like it is only my attacks that can cause it.  Any way of more reliably testing this?

Reply:
Thanks, yeah I wasn't sure about having it in cost, it's been moved into death_knight_action_t now.  Using howling blast, I also can't get it to proc on more than 1 mob per cast, fyi.

Also added a note that the damage reduction portion is not implemented yet.  Is there a way to build a debuff so that the damage reduction is only applied to the caster?  I'm not sure that it actually matters in practical terms for the sim, but wanted to check in to see if there was a way to do it to match in game behavior, before applying a blanket reduction.

Reply:
I think the only reliable way to check with DRW is if you observe a double spread from a single heart strike hit, so a >1 spread with one enemy affected, or >2 spread if your heart strike hits two enemies affected by shackles.

Reply:
For your second question: debuffs are based on actor pairs and as long as you get the right player's target data you should only see the debuffs that they applied themselves.